### PR TITLE
Broadcast

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
initial commit

### CODE REVIEW

File: widget/view.controller.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+
+        function init() {
+            if (_config.moduleType == 'Across Modules') { getTopXrecords(); }
+            else { getRecordsFromCustomModule(); }
+        }
+        init();
+
+        //check if this event is supposed to listen to eny of the broadcasted events, if yes then turn the listner up
+        if (_config.broadcastEvent) {
```
Comment (cs-bk) [inline widget/view.controller.js]: @cs-yash-patodkar - 29-30 in init

File: widget/view.controller.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                            element.style.opacity = 1;
+                        }, 600);
+                    })
+                }
+            })
+        }
+
+        function getTopXrecords() {
+            //building query
+            _config.query.sort = [{
```
Comment (cs-bk) [inline widget/view.controller.js]: @cs-yash-patodkar - make a separate function for getQuery

File: widget/view.controller.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                    'type': 'object',
+                    'value': 'false'
+                }
+            ]
+            var _queryObj = new Query(_config.query);
+
+            getResourceData(_config.module, _queryObj).then(function (result) {
+                var _dataSource = undefined;
+                if (result && result['hydra:member'] && result['hydra:member'].length > 0) {
+                    $scope.res = result['hydra:member'];
```
Comment (cs-bk) [inline widget/view.controller.js]: @cs-yash-patodkar - $scope.res seems to be local variable, if so no need to keep it in scope

File: widget/edit.html [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+                    <uib-accordion>
+                        <div uib-accordion-group class="queue-card-wrapper panel-default padding-top-lg">
+                            <uib-accordion-heading>
+                                <div class="queue-card-header padding-left-md workspace-connector-metadata-container"
+                                    ng-click="toggleArrow()" data-ng-class="{'queue-text-active-color': toggle">
+                                    <div class="margin-right-lg pull-left"><span class="queue-card-actions">&nbsp;<i
+                                                class="fa {{ toggle ? 'fa-chevron-down' : 'fa-chevron-right'}}"></i></span>
+                                    </div>
+                                    <div class="pull-left queue-card-header-text margin-right-3">Advanced Settings
```
Comment (cs-bk) [inline widget/edit.html:146]: Screen Shot 2023-07-27 at 9 43 10 AM
@cs-yash-patodkar - wrap advanced settings

File: widget/edit.html [hunk 5]
```diff
@@ -0,0 +1,183 @@
+<form data-ng-submit="save()" class="noMargin" name="editTopXWidgetForm" data-ng-class="{'state-wait': processing }"
```
Comment (cs-bk) [inline widget/edit.html:1]: @cs-yash-patodkar - Review spacing between with UX

File: widget/edit.controller.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        $scope.config = config;
+        $scope.onModuleChange = onModuleChange;
+        $scope.onChangeModuleType = onChangeModuleType;
+        $scope.moduleType = {
+            type: ['Across Modules', 'Single Module']
+        }
+        $scope.jsonObjModuleList=[];
+        $scope.config.eventName = $scope.config.eventName ? $scope.config.eventName : "";
+        $scope.config.broadcastEvent = $scope.config.broadcastEvent ? $scope.config.broadcastEvent : false;
+        $scope.toggleArrow = toggleArrow;
```
Comment (cs-bk) [inline widget/edit.controller.js]: @cs-yash-patodkar - better variable name

File: widget/view.controller.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+        //check if this event is supposed to listen to eny of the broadcasted events, if yes then turn the listner up
+        if (_config.broadcastEvent) {
+            //example widget:globalVisibilityEvent from Record Summary card
+            $rootScope.$on("widget:" + _config.eventName, function (event, data) {
+                var element = document.getElementById("topXParentDiv-" + _config.wid);
+                element.style.visibility = 'hidden';
+                element.style.opacity = 0;
+                element.style.transition = 'visibility 0.3s linear,opacity 0.3s linear';
+                if (_config.moduleType == 'Single Module') {
```
Comment (cs-bk) [inline widget/view.controller.js]: @cs-yash-patodkar - why no else condition?

**Review by cs-bk — APPROVED** [review]
LGTM

**Review by cs-suyog-jinturkar — APPROVED** [review]
BK already approved this PR, hence marking it as approved here as well.

**Pull request merged**